kinesisfirehose-logs-extension-demo/README.md

```diff
@@ -0,0 +1,226 @@
+
+> Note: This is a simple example extension to help you investigate an approach to centralize the log aggregation. This example code is not production ready. Use it with your own discretion after testing thoroughly.
+
+This sample extension:
+
+* Subscribes to receive `platform` and `function` logs.
+* Runs with a main, and a helper goroutine: The main goroutine registers to `ExtensionAPI` and process its `invoke` and `shutdown` events. The helper goroutine:
+  * starts a local HTTP server at the provided port (default 1234, the port can be overridden with Lambda environment variable `HTTP_LOGS_LISTENER_PORT` ) that receives requests from Logs API with `NextEvent` method call
+  * puts the logs in a synchronized queue (Producer) to be processed by the main goroutine (Consumer)
+* The main goroutine writes the received logs to Amazon Kinesis firehose, which gets stored in Amazon S3
+
+## Amazon Kinesis Data firehose
+
+Amazon Kinesis Data Firehose is the easiest way to reliably load streaming data into data lakes, data stores, and analytics services. It can capture, transform, and deliver streaming data to Amazon S3, Amazon Redshift, Amazon Elasticsearch Service, generic HTTP endpoints, and service providers like Datadog, New Relic, MongoDB, and Splunk, read more about it [here](https://aws.amazon.com/kinesis/data-firehose)
+
+> Note: The code sample provided part of this pattern delivers logs from Kinesis firehose to Amazon S3
+
+## Lambda extensions
+
+Lambda Extensions, a new way to easily integrate Lambda with your favorite monitoring, observability, security, and governance tools. Extensions are a new way for tools to integrate deeply into the Lambda environment. There is no complex installation or configuration, and this simplified experience makes it easier for you to use your preferred tools across your application portfolio today. You can use extensions for use-cases such as:
+
+* capturing diagnostic information before, during, and after function invocation
+* automatically instrumenting your code without needing code changes
+* fetching configuration settings or secrets before the function invocation
+* detecting and alerting on function activity through hardened security agents, which can run as separate processes from the function
+
+read more about it [here](https://aws.amazon.com/blogs/compute/introducing-aws-lambda-extensions-in-preview/)
+
+> Note: The code sample provided part of this pattern uses **external** extension to listen to log events from the Lambda function
+
+## Need to centralize log collection
+
+Having a centralized log collecting mechanism using Kinesis firehose provides the following benefits:
+
+* Helps to collect logs from different sources in one place. Even though the sample provided sends logs from Lambda, log routers like `Fluentbit` and `Firelens` can send logs directly to Kinesis Data firehose from container orchestrators like `EKS` and `ECS`.
```

Comment:
Kinesis Data Firehose

Reply:
Expand the service names on 1st mention so ppl know what they refer to [Amazon Elastic Kubernetes Service (EKS)]()" and "[Amazon Elastic Container Service (ECS)](https://aws.amazon.com/eks)"